doc/api/assert.markdown

```diff
@@ -3,9 +3,10 @@
     Stability: 3 - Locked
 
 The `assert` module provides a simple set of assertion tests that can be used
-to test invariants and implement unit tests. While the `assert` module is
-generally intended for internal use by Node.js itself, it can be used by user
-code calling `require('assert')`.
+to test invariants and implement unit tests. It is recommended that users of
+Node.js choose a third party assertion library from [npm][] instead of relying
+on `assert`.  That being said, `assert` can be used by user code calling
```

Comment:
@mikeal This statement should not be this strong. There is _no_ commonly npm package in existence for asserting ***invariants*** in source. At least, not that I know of, can you point to one?

I suggest we make clear that assert is NOT for unit tests, and not point people to non-existent modules for the use-case that assert is a good tool for (invariants).

> The `assert` module provides a simple set of assertion tests that can be used
 to test invariants. While the `assert` module is generally intended for internal use by Node.js itself, it can be used by user code calling `require('assert')`. For the purposes of unit testing, it is recommended that users of Node.js choose a third party test library from [npm][] instead of relying on `assert`.